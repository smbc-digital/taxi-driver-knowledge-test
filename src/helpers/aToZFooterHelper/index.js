const showAToZFooter = (show) =>{
    let footer  = document.getElementById('atoz')
    if(footer !== null) 
    {
        if(show){
            footer.classList.remove('hidden')
        }
        else{
            footer.classList.add('hidden')
        }
    }
}

export default showAToZFooter