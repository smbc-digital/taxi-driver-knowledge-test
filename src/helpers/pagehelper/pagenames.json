[
	{
		"number": 1,
		"route": "/book-taxi-driver-knowledge-test/test-type"
	},
	{
		"number": 2,
		"route": "/book-taxi-driver-knowledge-test/first-time"
	},
	{
		"number": 3,
		"route": "/book-taxi-driver-knowledge-test/about-yourself"
	},
	{
		"number": 4,
		"route": "/book-taxi-driver-knowledge-test/choose-appointment"
	},
	{
		"number": 5,
		"route": "/book-taxi-driver-knowledge-test/booking-summary"
	},
	{
		"number": 6,
		"route": "/book-taxi-driver-knowledge-test/success"
	},
	{
		"number": 7,
		"route": "/book-taxi-driver-knowledge-test/something-went-wrong"
	},
	{
		"number": 8,
		"route": "/book-taxi-driver-knowledge-test/failed-booking"
	},
	{
		"number": 9,
		"route": "/book-taxi-driver-knowledge-test/payment-declined"
	}
]