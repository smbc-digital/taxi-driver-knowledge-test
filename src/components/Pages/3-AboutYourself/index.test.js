import { React, mount } from '../../../helpers/SetupTest'
import { AboutYourself } from './index'
import { getPageRoute } from '../../../helpers/pagehelper'
import renderer from 'react-test-renderer'

describe('AboutYourself', () => {
    it('should call push on submit', () => {
        //Arrange
        const data = {
            firstName: {
                value:'Sarah',
                isValid: true
            },
            lastName: {
                value:'Logan',
                isValid: true
            },
            phoneNumber: {
                value:'Value',
                isValid: true
            },
            emailAddress: {
                value:'Value',
                isValid: true
            },
            address: {
                value:'Value',
                isValid: true
            },
            onChange: null
        }

        const history = { push: jest.fn() }
        const wrapper = mount(<AboutYourself context={data} history={history} />)

        //Act
        wrapper.find('form').simulate('submit')

        //Assert
        expect(history.push).toHaveBeenCalledWith(getPageRoute(4))
    })

    describe('snapshot', () => {
        it('should render correctly', () => {
            //Arrange
            const data = {
                firstName: {
                    value:'Sarah',
                    isValid: true
                },
                lastName: {
                    value:'Logan',
                    isValid: true
                },
                phoneNumber: {
                    value:'Value',
                    isValid: true
                },
                emailAddress: {
                    value:'Value',
                    isValid: true
                },
                address: {
                    value:'Value',
                    isValid: true
                },
                onChange: null
            }

            // Act
			const tree = renderer
			.create(<AboutYourself context={data} />)
			.toJSON()
		
			// Assert
			expect(tree).toMatchSnapshot()

        })
    })
})