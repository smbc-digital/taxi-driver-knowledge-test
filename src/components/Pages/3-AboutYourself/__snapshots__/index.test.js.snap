// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutYourself snapshot should render correctly 1`] = `
Array [
  <form
    onSubmit={[Function]}
  >
    <h1 />
    <h2>
      Tell us about yourself
    </h2>
    <label
      className=""
      htmlFor="firstName"
    >
      First name
    </label>
    <input
      aria-label="First name"
      aria-live="off"
      className=""
      id="firstName"
      inputMode={undefined}
      maxLength="35"
      min={undefined}
      name="firstName"
      onChange={[Function]}
      onKeyDown={[Function]}
      pattern={undefined}
      type="text"
      value="Sarah"
    />
    <label
      className=""
      htmlFor="lastName"
    >
      Last name
    </label>
    <input
      aria-label="Last name"
      aria-live="off"
      className=""
      id="lastName"
      inputMode={undefined}
      maxLength="60"
      min={undefined}
      name="lastName"
      onChange={[Function]}
      onKeyDown={[Function]}
      pattern={undefined}
      type="text"
      value="Logan"
    />
    <label
      className=""
      htmlFor="phoneNumber"
    >
      Phone number
    </label>
    <input
      aria-label="Phone number"
      aria-live="off"
      className=""
      id="phoneNumber"
      inputMode={undefined}
      maxLength={undefined}
      min={undefined}
      name="phoneNumber"
      onChange={[Function]}
      onKeyDown={[Function]}
      pattern={undefined}
      type="tel"
      value="Value"
    />
    <label
      className=""
      htmlFor="emailAddress"
    >
      Email address
    </label>
    <input
      aria-label="Email address"
      aria-live="off"
      className=""
      id="emailAddress"
      inputMode={undefined}
      maxLength={undefined}
      min={undefined}
      name="emailAddress"
      onChange={[Function]}
      onKeyDown={[Function]}
      pattern={undefined}
      type="email"
      value="Value"
    />
    <label
      className=""
      htmlFor="addressLine1"
    >
      Address line 1
    </label>
    <input
      aria-label="Address line 1"
      aria-live="off"
      className=""
      id="addressLine1"
      inputMode={undefined}
      maxLength={undefined}
      min={undefined}
      name="addressLine1"
      onChange={[Function]}
      onKeyDown={[Function]}
      pattern={undefined}
      type="text"
      value=""
    />
    <h3
      aria-hidden="true"
      className="optional"
    >
      Address line 2
    </h3>
    <p
      className="optional"
    >
      (optional)
    </p>
    <label
      className="optional sr-only"
      htmlFor="addressLine2"
    >
      Address line 2 optional
    </label>
    <input
      aria-label=""
      aria-live=""
      className=""
      id="addressLine2"
      inputMode={undefined}
      maxLength={undefined}
      min={undefined}
      name="addressLine2"
      onChange={[Function]}
      onKeyDown={[Function]}
      pattern={undefined}
      type="text"
      value=""
    />
    <label
      className=""
      htmlFor="town"
    >
      Town
    </label>
    <input
      aria-label="Town"
      aria-live="off"
      className=""
      id="town"
      inputMode={undefined}
      maxLength={undefined}
      min={undefined}
      name="town"
      onChange={[Function]}
      onKeyDown={[Function]}
      pattern={undefined}
      type="text"
      value=""
    />
    <label
      className=""
      htmlFor="postcode"
    >
      Postcode
    </label>
    <input
      aria-label="Postcode"
      aria-live="off"
      className=""
      id="postcode"
      inputMode={undefined}
      maxLength={undefined}
      min={undefined}
      name="postcode"
      onChange={[Function]}
      onKeyDown={[Function]}
      pattern={undefined}
      type="text"
      value=""
    />
    <p>
      <a
        href="#"
        onClick={[Function]}
      >
         Use address finder 
      </a>
    </p>
    <button
      className="button-primary"
      disabled={false}
      id={undefined}
      type="submit"
    >
      Next step
    </button>
  </form>,
  <p>
    <a
      href="#"
      onClick={[Function]}
    >
      Back
    </a>
  </p>,
]
`;
